src/client.js

```diff
@@ -218,26 +219,18 @@ assign(Client.prototype, {
 
   // Acquire a connection from the pool.
   acquireConnection() {
-    let request = null
-    const completed = new Promise((resolver, rejecter) => {
-      if (!this.pool) {
-        return rejecter(new Error('Unable to acquire a connection'))
+    if (!this.pool) {
+      return {
+        completed: Promise.reject(new Error('Unable to acquire a connection'))
       }
-      request = this.pool.acquire(function(err, connection) {
-        if (err) return rejecter(err)
-        debug('acquired connection from pool: %s', connection.__knexUid)
-        resolver(connection)
-      })
-    })
-    const abort = function(reason) {
-      if (request && !request.fulfilled) {
-        request.abort(reason)
+    } else {
+      return {
+        completed: this.pool.acquire().promise.then((connection) => {
+          debug('acquired connection from pool: %s', connection.__knexUid)
+          return connection
+        })
```

Comment:
Tarn handles the acquire timeout. There is no need to handle it on the knex side anymore. There are tests for this.